internal/signature/key.go

```diff
@@ -4,22 +4,22 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"regexp"
 
 	"github.com/Azure/notation-azure-kv/internal/cloud"
 	"github.com/notaryproject/notation-core-go/signature"
 	"github.com/notaryproject/notation-go/plugin/proto"
 )
 
-func newKey(keyID string, pluginConfig map[string]string) (*cloud.Key, error) {
-	client, err := cloud.NewAzureClient()
+// re is the regular expression for parsing keyID: https://{vaultName}.vault.azure.net/keys/{keyName}/{version}
+const re = `^https:\/\/(.*?)\.(.*)\/keys\/(.*)\/(.*)`
```

Comment:
The original `url.Parse()` + `strings.Split()` way is preferred and reduces bugs.

Note: The invalid `https://<name>.vault.azure.net/keys/<keyname>/<version>/error` is accepted by the above regex expression.